providers/ringcentral/src/lib/ringcentral.provider.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+
+import '../config';
+import { SDK as RingCentralSDK } from '@ringcentral/sdk';
+import Platform from '@ringcentral/sdk/lib/platform/Platform';
+
+export class RingCentralSmsProvider implements ISmsProvider {
+  id = 'ringcentral';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  private rcsdk: RingCentralSDK;
+  private platform: Platform;
+
+  constructor(
+    private config: {
+      username: string;
+      password: string;
+      from: string;
+    }
+  ) {
+    this.rcsdk = new RingCentralSDK({ server: process.env.RC_SERVER_URL, clientSecret: process.env.RC_CLIENT_SECRET, clientId: process.env.RC_CLIENT_ID });
```

Comment:
Unfortatently we can't use env variables to those params, they should be passed in the constructor since we need to change it dynamically during runtime